### DISCUSSION TITLE
Pre-recorded audio transcription in Unity

### DISCUSSION THREAD

**Discussion by SeaTiger11** [deepgram/community#496] [General help]
For our team's project we intend to have the audio recorded locally in Unity then transcribed all at once and I believe that the pre-recorded system would be more ideal for this process compared to the live streaming example. Unfortunately, there's no guide to implementing pre-recorded into Unity and the documentation is unclear as to how you'd implement it into Unity. If anyone can offer advice or tips to getting it set up, that would be greatly appreciated.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment]
To do a pre-recorded DG request from Unity would require doing an HTTP POST of the binary audio - it looks like Unity has some built-in functions to do POST requests - https://docs.unity3d.com/ScriptReference/Networking.UnityWebRequest.Post.html - it is not yet clear to me if one can specify the body as binary though (but I'm finding some forum posts which seem to be showing how to do this). If this helps, great, if not, I'll try to code up an example on Wednesday!

**Reply by SeaTiger11** [reply]
I've since created a working system for pre-recorded audio using urls however my implementation will require uploading byte arrays of audio and I'm currently unsure of how to format it. Would you happen to know the correct was to format it? (I've been basing the other data fields such as headers and so on from the curl example code for Whisper however there's only code for sending audio based on urls).
